src/components/common/state-machine/state-machine.ts

```diff
@@ -131,6 +131,10 @@ const authStateMachine = createMachine(
               cond: "isReauthenticationRequired",
             },
             { target: [PATH_NAMES.AUTH_CODE], cond: "isAuthenticated" },
+            {
+              target: [PATH_NAMES.ENTER_PASSWORD],
```

Comment:
This condition will have no effect since the condition two above this will take precedence. (Did you write a test to check this did what you expected it to? This would be a good thing to TDD) I think instead we should be changing the `TARGET` in the condition two above this to go to `ENTER_PASSWORD` instead of `SIGN_IN_OR_CREATE`. There are tests already, this would be a good thing to TDD.